### ISSUE TITLE
Test failure - test_segment_from_mask_non_square

### ISSUE THREAD

**Issue by GenevieveBuckley** [computational-cell-analytics/micro-sam#371]
There is a failing test: test_segment_from_mask_non_square

It fails when you run the tests from either the master or dev branches.
* [Here are the results from the master branch](https://github.com/computational-cell-analytics/micro-sam/actions/runs/7722147756)
* [Same results from the dev branch](https://github.com/computational-cell-analytics/micro-sam/actions/runs/7722146152)

Here are the relevant sections of test code:
* [the test that fails](https://github.com/computational-cell-analytics/micro-sam/blob/8151561ea005331ed34b5d691c7440e79db4db7c/test/test_prompt_based_segmentation.py#L106-L107)
* [the actual part that runs](https://github.com/computational-cell-analytics/micro-sam/blob/8151561ea005331ed34b5d691c7440e79db4db7c/test/test_prompt_based_segmentation.py#L52-L101)
* [the line reported as failing](https://github.com/computational-cell-analytics/micro-sam/blame/8151561ea005331ed34b5d691c7440e79db4db7c/test/test_prompt_based_segmentation.py#L77)

This is the error message:

 FAILED test/test_prompt_based_segmentation.py::TestPromptBasedSegmentation::test_segment_from_mask_non_square - AssertionError: 0.012674722835867252 not greater than 0.8

```python-traceback
=================================== FAILURES ===================================
________ TestPromptBasedSegmentation.test_segment_from_mask_non_square _________

self = <test.test_prompt_based_segmentation.TestPromptBasedSegmentation testMethod=test_segment_from_mask_non_square>

    def test_segment_from_mask_non_square(self):
>       self._test_segment_from_mask((256, 384))

test/test_prompt_based_segmentation.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/test_prompt_based_segmentation.py:77: in _test_segment_from_mask
    self.assertGreater(util.compute_iou(mask, predicted), expected_iou_mask)
E   AssertionError: 0.012674722835867252 not greater than 0.8
```

**Comment by GenevieveBuckley** [comment]
It looks like @constantinpape wrote this particular test, so perhaps he's the best person to ask what it's supposed to check here?

**Comment by constantinpape** [comment]
Thanks for opening the issue @GenevieveBuckley. I will look into this.

**Comment by constantinpape** [comment]
I figured out the issue: the behavior of `ResizeLongestSide.apply_image` has changed. This now casts the output to `uint8` and changes the data range. Before it left data type and range unchanged. My best guess is that this change was introduced by the recent upgrade of PyTorch / torchvision.

For the concrete test the issue stems from [here](https://github.com/computational-cell-analytics/micro-sam/blob/master/micro_sam/prompt_based_segmentation.py#L95).
It should be rather easy to fix this behavior here, but we rely on this in a couple of other places, where changes might also be needed.

**Comment by constantinpape** [comment]
I checked, this is the only place affected. Will be fixed by #375.

**Comment by GenevieveBuckley** [comment]
Nice, thank you!
